hack/build-k8s-matrix.sh

```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+set -e
+
+start_minor=22 # TODO: lower this to ~10
```

Comment:
this is used to support sever-side apply starting from 1.22?

Reply:
It's just an arbitrary version that is currently passing. I'll send another PR to fix older versions and update this value.